### PR TITLE
steam: fix crash at startup with open source radeon drivers

### PR DESCRIPTION
**Pull request by corngood**
###### Motivation for this change

This is basically a port of [this fix from OpenSUSE](https://bugzilla.opensuse.org/show_bug.cgi?id=988273).

Steam ships statically linked with a version of libcrypto that conflics with the one dynamically loaded by radeonsi_dri.so.

This provides a compatible one (w/ engine disabled, cpuid fix) and links it into the steam runtime.

The error looks like:

```
steam.sh: line 713: 7842 Segmentation fault (core dumped) 
```

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8702,7 +8702,8 @@ in
     })
     openssl_1_0_1
     openssl_1_0_2
-    openssl_1_1_0;
+    openssl_1_1_0
+    openssl_1_0_2-steam;
```
Comment (rasendubi) [inline pkgs/top-level/all-packages.nix]: We can add `openssl-steam = openssl_1_0_2-steam;` above and use that name in the steam package.

Comment (corngood) [inline pkgs/top-level/all-packages.nix]: Yeah, good call.  Updated.

**Review by fpletz — APPROVED** [review]

**Review by rasendubi — APPROVED** [review]

**Comment by FRidh** [comment]
This broke the game "Wargame: European Escalation" for me.
```
relocation error: /home/freddy/.local/share/Steam/steamapps/common/Wargame European Escalation/Wargame: symbol ENGINE_load_builtin_engines, version OPENSSL_1.0.0 not defined in file libcrypto.so.1.0.0 with link time reference
```

**Comment by corngood** [comment]
@FRidh Could you open an issue for this and tag me on it?

It looks like ubuntu has a version-script.patch for openssl which adds these version symbols.

I can try to build openssl-steam with that patch, but I might need your help to test it.

**Pull request merged**